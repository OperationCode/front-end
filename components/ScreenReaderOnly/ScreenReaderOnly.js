import React from 'react';
import { node } from 'prop-types';
import { SCREEN_READER_ONLY } from 'common/constants/testIDs';
import styles from './ScreenReaderOnly.module.css';

export const toggleMessages = {
  open: 'Show more',
  close: 'Hide expanded',
};

ScreenReaderOnly.propTypes = {
  children: node.isRequired,
};

function ScreenReaderOnly({ children }) {
  return (
    <span className={styles.ScreenReaderOnly} data-testid={SCREEN_READER_ONLY}>
      {children}
    </span>
  );
}

export default ScreenReaderOnly;
