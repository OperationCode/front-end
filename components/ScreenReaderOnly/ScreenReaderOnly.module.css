.ScreenReaderOnly {
  clip: rect(0, 0, 0, 0);
}
