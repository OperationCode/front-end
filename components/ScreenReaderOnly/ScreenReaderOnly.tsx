import { SCREEN_READER_ONLY } from '@/common/constants/testIDs';

export const toggleMessages = {
  open: 'Show more',
  close: 'Hide expanded',
};

export function ScreenReaderOnly({ children }: React.PropsWithChildren<unknown>) {
  return (
    <span className="sr-only" data-testid={SCREEN_READER_ONLY}>
      {children}
    </span>
  );
}
