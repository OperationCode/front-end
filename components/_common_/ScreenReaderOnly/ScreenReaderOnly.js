import React from 'react';
import PropTypes from 'prop-types';
import styles from './ScreenReaderOnly.css';

ScreenReaderOnly.propTypes = {
  children: PropTypes.string.isRequired,
};

export default function ScreenReaderOnly({ children }) {
  return <span className={styles.ScreenReaderOnly}>{children}</span>;
}
