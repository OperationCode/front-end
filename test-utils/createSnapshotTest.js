/* eslint-env jest */
import renderer from 'react-test-renderer';

/**
 * Used to create fully-rendered snapshot test.
 * Useful for reusable component smoke tests.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();
};
