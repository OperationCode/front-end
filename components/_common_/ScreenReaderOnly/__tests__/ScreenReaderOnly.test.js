/* eslint-env jest */
import React from 'react';
import createSnapshotTest from 'test-utils/createSnapshotTest';

import ScreenReaderOnly from '../ScreenReaderOnly';

describe('ScreenReaderOnly', () => {
  it('should render with required props', () => {
    createSnapshotTest(<ScreenReaderOnly>Test</ScreenReaderOnly>);
  });
});
