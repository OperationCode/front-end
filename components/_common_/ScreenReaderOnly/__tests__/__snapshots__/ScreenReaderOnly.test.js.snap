// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenReaderOnly should render with required props 1`] = `
<span
  className="ScreenReaderOnly"
>
  Test
</span>
`;
